### PR TITLE
Fixed soapHeader addition for both server and client

### PR DESCRIPTION
**Pull request by rashmihunt**
Fixed soapHeader addition for both server and client.

### CODE REVIEW

File: src/base.js [hunk 1]
```diff
@@ -122,6 +123,28 @@ class Base extends EventEmitter {
     return nsContext;
   }
 
+  addSoapHeadersToEnvelope(soapHeaderElement, xmlHandler) {
+    for (let i = 0, n = this.soapHeaders.length; i < n; i++) {
+      let soapHeader = this.soapHeaders[i];
+      let elementDescriptor;
+      if (typeof soapHeader.value === 'object') {
+        if (soapHeader.qname !== undefined) {
+          if (soapHeader.qname.nsURI !== undefined) {
```
Comment (raymondfeng) [inline src/base.js]: Nitpick: we can use `if(soapHeader.qname && soapHeader.qname.nsURI)`

File: src/base.js [hunk 2]
```diff
@@ -122,6 +123,28 @@ class Base extends EventEmitter {
...
+            let element = this.findElement(soapHeader.qname.nsURI, soapHeader.qname.name);
+            elementDescriptor =
+              element && element.describe(this.wsdl.definitions);
+          }
+        }
+        xmlHandler.jsonToXml(soapHeaderElement, null, elementDescriptor,
+          soapHeader.value);
+      } else { //soapHeader has XML value
+        XMLHandler.parseXml(soapHeaderElement, soapHeader.xml);
+        xmlHandler.jsonToXml(soapHeaderElement, null, null,
```
Comment (raymondfeng) [inline src/base.js]: The parseXml already adds the xml string into the soapHeaderElement. I don't see the need to call xmlHandler.jsonToXml again unless we allow both xml and value.

File: src/parser/xmlHandler.js [hunk 3]
```diff
@@ -118,7 +118,8 @@ class XMLHandler {
   mapObject(node, nsContext, descriptor, val) {
     if (val == null) return node;
     if (typeof val !== 'object' || (val instanceof Date)) {
-      node.text(val);
+      parseXml(node, val, null);
```
Comment (raymondfeng) [inline src/parser/xmlHandler.js]: I'm not sure if this logic is correct unless parseXml can deal with non-xml string or date.

File: src/base.js (hunk 2 above)
Comment (rashmihunt) [inline src/base.js]: Fixed

File: src/parser/xmlHandler.js (hunk 3 above)
Comment (rashmihunt) [inline src/parser/xmlHandler.js]: Fixed. Unintentionally left out parseXML line while trying out correct execution point to convert soapHeader to xml. It's done in base.js for both client and server.

**Comment by raymondfeng** [comment]
LGTM

**Pull request merged**